src/shims/unix/android/foreign_items.rs

```diff
@@ -25,8 +26,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(errno_place.to_ref(this).to_scalar(), dest)?;
             }
 
+            // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `prctl` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                check_args_len("prctl", args, 1)?;
+
+                let id = this.read_scalar(&args[0])?.to_i32()?;
+                // FIXME: Use PR_SET_NAME and PR_GET_NAME constants when
+                // https://github.com/rust-lang/libc/pull/3941 lands.
+                let pr_set_name = 15_i32;
+                let pr_get_name = 16_i32;
+
+                let res = match id {
+                    id if id == pr_set_name => {
+                        check_args_len("'PR_SET_NAME' prctl", args, 2)?;
+
+                        let tid = this.linux_gettid()?;
+                        let name = this.read_scalar(&args[1])?;
+                        let name_len = 16;
+
+                        this.pthread_setname_np(tid, name, name_len)?
+                    }
+                    id if id == pr_get_name => {
+                        check_args_len("'PR_GET_NAME' prctl", args, 2)?;
+
+                        let tid = this.linux_gettid()?;
+                        let name = this.read_scalar(&args[1])?;
+                        let name_len = Scalar::from_target_usize(16, this);
+
+                        this.pthread_getname_np(tid, name, name_len)?
+                    }
+                    id => {
+                        this.handle_unsupported_foreign_item(format!(
+                            "can't execute prctl with ID {id}"
+                        ))?;
+                        return Ok(EmulateItemResult::AlreadyJumped);
+                    }
+                };
+                this.write_scalar(res, dest)?;
+            }
+
             _ => return Ok(EmulateItemResult::NotSupported),
         }
         Ok(EmulateItemResult::NeedsReturn)
     }
 }
+
+fn check_args_len<'tcx>(
+    link_name: &str,
+    args: &[OpTy<'tcx>],
+    args_expected: usize,
+) -> InterpResult<'tcx, ()> {
+    let args_actual = args.len();
+    if args_actual < args_expected {
+        throw_ub_format!(
+            "incorrect number of arguments for {link_name}: got {args_actual}, expected at least {args_expected}"
+        );
```

Comment:
I know that there are a lot of place using ``throw_ub_format!`` when the number of arguments is incorrect, so I am wondering if there is a consensus on when to use ``throw_unsup_format`` vs ``throw_ub_format``,because an incorrect number of arguments is not the kind of [Undefined Behaviour](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) I am thinking when it comes to miri.

Reply:
It is definitely UB to call a function with fewer arguments than it expects, and Miri should report this as such.

Reply:
Switched back to `throw_ub_format`.